source/app-development/tutorial-ps-to-quota.rst

```diff
@@ -0,0 +1,451 @@
+      # AppProcess objects, one per process
+      def parse(output)
+        lines = output.strip.split("\n")
+        lines.drop(2).map do |line|
+          Quota.new(*(line.split))
+        end
+      end
+
+Now when we run the test they pass:
+
+.. code:: sh
+
+    $ rake
+    Run options: --seed 60317
+
+    # Running:
+
+    .
+
+    Finished in 0.000966s, 1035.1494 runs/s, 6210.8963 assertions/s.
+
+    1 runs, 6 assertions, 0 failures, 0 errors, 0 skips
+
+Update app.rb and view/index.html
+.................................
+
+Update app.rb:
+
+.. code:: diff
+
+    helpers do
+      def title
+    -    "Passenger App Processes"
+    +    "Quota"
+      end
+    end
+
+    # Define a route at the root '/' of the app.
+    get '/' do
+      @command = Command.new
+    -  @processes, @error = @command.exec
+    +  @quotas, @error = @command.exec
+
+      # Render the view
+      erb :index
+    end
+
+
+In view/index.html, replace the table with this:
+
+.. code:: erb
+
+    <table class="table table-bordered">
+      <tr>
+        <th>Filesystem</th>
+        <th>Blocks</th>
+        <th>Blocks Quota</th>
+        <th>Blocks Limit</th>
+        <th>Blocks Grace</th>
+        <th>Files</th>
+        <th>Files Quota</th>
+        <th>Files Limit</th>
+        <th>Files Grace</th>
+      </tr>
+      <% @quotas.each do |quota| %>
+      <tr>
+        <td><%= quota.filesystem %></td>
+        <td><%= quota.blocks %></td>
+        <td><%= quota.blocks_quota %></td>
+        <td><%= quota.blocks_limit %></td>
+        <td><%= quota.blocks_grace %></td>
+        <td><%= quota.files %></td>
+        <td><%= quota.files_quota %></td>
+        <td><%= quota.files_limit %></td>
+        <td><%= quota.files_grace %></td>
+      </tr>
+      <% end %>
+    </table>
+
+These changes should not require an app restart. Go to the launched app and reload the page to see the changes.
+
+Brand App
+---------
+
+The app is looking good, but the details page still shows the app title "Passenger App Processes". To change this and the icon, edit the manifest.yml:
+
+.. code:: diff
+
+    -name: Passenger App Processes
+    -description: Display your running Passenger app proceseses in a table
+    +name: Quota
+    +description: Display quotas
+    +icon: fa://hdd-o
+
+* The icon follows format of ``fa://{FONTAWESOMENAME}`` where you replace ``{FONTAWESOMENAME}`` with an icon from http://fontawesome.io/icons/.
+  In this case we are using ``fa-hdd-o`` which we write in the manifest as ``fa://hdd-o``: http://fontawesome.io/icon/hdd-o/
+
+Publish App
+-----------
+
+Publishing an app requires two steps:
+
+#. Updating the manifest.yml to specify the category and optionally subcategory, which indicates where in the dashboard menu the app appears.
+#. Having an administrator checkout a copy of the production version to a directory under /var/www/ood/apps/sys
+
+
+Steps:
+
+#. Add category to manifest so app appears in Files menu:
+
+    .. code:: diff
```

Comment:
Adother diff, it looks like instead of using `code::`, sphinx has a `:diff:` option. All of the diffs should be replaced with this if it works.

See  http://www.sphinx-doc.org/en/stable/markup/code.html